### PR TITLE
[BUG] 651: implements strtok_to_array function

### LINKED ISSUE #651: [BUG] Missing function: strtok_to_array
**Function:** `strtok_to_array`
**Category:** String & binary
**Status:** Failed (Return Code: 1)
**Documentation:** https://docs.snowflake.com/en/sql-reference/functions/strtok_to_array

**Test SQL:**
```sql
SELECT STRTOK_TO_ARRAY('snow-flake', '-');
```

**Expected Result:**
```
N/A
```

**Actual Result:**
```
SELECT STRTOK_TO_ARRAY('snow-flake', '-');
```

**Stderr:**
```
╭─ Error ──────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ 000200: 200: DataFusion error: Error during planning: Invalid function 'strtok_to_array'.                    │
│ Did you mean 'string_to_array'?                                                                              │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
```

**Parent EPIC:** #576

### PR DESCRIPTION
**Pull request by AmosAidoo**
## Which issue does this PR close?

- Closes #651 

## Changes

- Added a new scalar udf, `StrtokToArrayFunc` and implemented logic according to snowflake docs

## How was this tested?

- Added automated tests to validate functionality
- All existing tests pass

### CODE REVIEW

**Comment by AmosAidoo** [comment]
@Vedin could you kindly review this PR when you have some time

File: crates/df-builtins/src/strtok_to_array.rs [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+impl Default for StrtokToArrayFunc {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl StrtokToArrayFunc {
+    pub fn new() -> Self {
+        Self {
+            signature: Signature::variadic(vec![DataType::Utf8], Volatility::Immutable),
```
Comment (Vedin) [inline crates/df-builtins/src/strtok_to_array.rs]: I believe `LargeUtf8` and `Utf8View` should be also acceptable here. In general, I'd suggest using `Signature::string` in functions that accept all types of strings. As I understood, it's always just 1 or 2 strings as arguments. So, you don't need to use `variadic` here. It can be replaced with `Signature::one_of` and specifying 2 signatures inside.

File: crates/df-builtins/src/strtok_to_array.rs [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+                "+-------------------+",
+                "| default_delimeter |",
+                "+-------------------+",
+                "| [a, b, c]         |",
+                "+-------------------+",
+            ],
+            &result
+        );
+
+        let q = "SELECT STRTOK_TO_ARRAY('a b c', NULL) AS null_result;";
```
Comment (Vedin) [inline crates/df-builtins/src/strtok_to_array.rs:192]: Please add a test when the first argument is NULL as well.

Comment (Vedin) [inline crates/df-builtins/src/strtok_to_array.rs:192]: I'm also not sure about what happens if we have JSON null here. E.G.
`  SELECT STRTOK_TO_ARRAY(PARSE_JSON('null'), '.@') AS multiple_delimiters;`
Let's test it as well.

**Review by Vedin — CHANGES_REQUESTED** [review]

File: crates/df-builtins/src/strtok_to_array.rs (hunk 2 above)
Comment (AmosAidoo) [reply crates/df-builtins/src/strtok_to_array.rs:192]: Thank you. I've made the changes and updated the tests.

**Review by Vedin — APPROVED** [review]

**Comment by AmosAidoo** [comment]
Not sure why that compile error(`Source path does not exist: /home/runner/work/embucket/embucket/ui/dist`) occured.
I have rebased the branch

**Comment by DanCodedThis** [comment]
> Not sure why that compile error(`Source path does not exist: /home/runner/work/embucket/embucket/ui/dist`) occured. I have rebased the branch

@AmosAidoo We have a CI flow to generate the UI now (as tar), you may just not have the .tar file. There were some problems with the path, but the latest main + having the `dist.tar` fixed it. It should work correctly, if not please ping @nikitastryuk and/or @YaroslavLitvinov

**Comment by YaroslavLitvinov** [comment]
> > Not sure why that compile error(`Source path does not exist: /home/runner/work/embucket/embucket/ui/dist`) occured. I have rebased the branch
> 
> @AmosAidoo We have a CI flow to generate the UI now (as tar), you may just not have the .tar file. There were some problems with the path, but the latest main + having the `dist.tar` fixed it. It should work correctley, if not pls ping @nikitastryuk and/or @YaroslavLitvinov

I'm looking into it, so far approved workflows to run as I don't see an actual error

**Comment by YaroslavLitvinov** [comment]
> Not sure why that compile error(`Source path does not exist: /home/runner/work/embucket/embucket/ui/dist`) occured. I have rebased the branch

As I see rebase helped.
